### PR TITLE
Improve Documentation Content

### PR DESCRIPTION
**Pull request by rob-shalloo**
Populated and improved several pages. Specifically, I've updated the landing page to include a basic description of the code, added a disclaimer about the development phase of the code and updated some text.
I've overhauled the `user guide` section and included updated installation instructions together with a `first example` to help people get started. Finally, I've added some information to the `code overview` section to attempt to explain how the code functions and what its structure is. This last bit might need some tlc at some stage, but I think its probably enough for the moment.

### CODE REVIEW

File: docs/source/code_overview/codes_supporting_lasy.rst [hunk 1]
```diff
@@ -1,4 +1,17 @@
...
-A list of codes which support LASY Data as input.
+A list of codes which support LASY Data as input:
+
+* Many Coming Soon!
+
+A list of codes currently in the process of adding support:
+
+* `FBPIC <https://github.com/fbpic/fbpic>`_
+* `Wake-T <https://github.com/AngelFP/Wake-T>`_
+* `WarpX <https://github.com/ECP-WarpX/WarpX>`_
```
Comment (RemiLehe) [inline docs/source/code_overview/codes_supporting_lasy.rst:12]: Should we add HiPACE++ ? (support for `lasy` is complete in this case, in my understanding)

File: docs/source/code_overview/index.rst [hunk 2]
```diff
@@ -1,13 +1,55 @@
 Overview of the Code
 ====================
 
-What's this all about then?
+``lasy`` manipulates laser pulses, and operates on the laser envelope.
+It can be used to define the 3D profile of a laser pulse.
+The user can define seperately the transverse and longitudinal profile of the laser pulse either from a range common analytic profiles or using experimental measurements.
+Once defined the laser pulse may be propagated to a user defined location.
+Finally the laser profile may outputed to file for use as an input to a variety of different simulation tools.
```
Comment (RemiLehe) [inline docs/source/code_overview/index.rst]:
```suggestion
Finally the laser profile may be outputted to file for use as an input to a variety of different simulation tools.
```

File: docs/source/index.rst [hunk 3]
```diff
@@ -6,22 +6,39 @@
...
     :card: + intro-card text-center
     :column: col-lg-6 col-md-6 col-sm-6 col-xs-12 p-2 d-flex
 
     ---
 
-    **User Guide**
+    **Getting Started**
     ^^^^^^^^^^^^^^
 
-    Instructions on getting started with the code.
```
Comment (RemiLehe) [inline docs/source/index.rst:18]: Could you fix the merge conflict below?

File: docs/source/index.rst [hunk 4]
```diff
@@ -43,7 +60,11 @@ LASY |release| Documentation
     **API Reference**
     ^^^^^^^^^^^^^^^^^
 
-    Detailed documentation of the LASY API.
```
Comment (RemiLehe) [inline docs/source/index.rst:46]: Another merge conflict to be fixed.

File: docs/source/user_guide/index.rst [hunk 5]
```diff
@@ -1,10 +1,67 @@
...
+    :caption: Installation Instructions
+
+    git clone https://github.com/LASY-org/lasy.git
+    cd lasy
+    python3 -m pip install -v .
+
+More installation options and further instructions will be added in due course.
+
+
+First Example
```
Comment (RemiLehe) [inline docs/source/user_guide/index.rst:20]: Nice. We should discuss again how to automatically test this.

**Comment by AngelFP** [comment]
From the discussion in the weekly meeting, what I use in Wake-T for formatted tutorials with text and code is `sphinx_gallery`. See an example [here](https://wake-t.readthedocs.io/en/latest/tutorials/00_basic_tutorial.html), and the source python file [here](https://github.com/AngelFP/Wake-T/blob/dev/tutorials/00_basic_tutorial.py).

File: docs/source/code_overview/codes_supporting_lasy.rst (hunk 1 above)
Comment (MaxThevenet) [reply docs/source/code_overview/codes_supporting_lasy.rst:12]: Yup, I think we should, thanks! Further tests to be done soon.

File: docs/source/index.rst [hunk 6]
```diff
@@ -6,22 +6,39 @@
...
-    **User Guide**
+    **Getting Started**
     ^^^^^^^^^^^^^^
 
-    Instructions on getting started with the code.
+<<<<<<< HEAD
+    New to ``lasy``? Check this out for installation instructions and a first example.
+=======
+    New to :mod:`lasy`? Check this out for installation instructions and a first example.
+>>>>>>> 3520659412e87dd3d7278db6c5fd8262a521017b
```
Comment (MaxThevenet) [inline docs/source/index.rst]: Uh oh, some merge conflicts made it to the PR, we could have a chat offline.

File: docs/source/code_overview/codes_supporting_lasy.rst [hunk 7]
```diff
@@ -1,4 +1,17 @@
...
+
```
Comment (MaxThevenet) [inline docs/source/code_overview/codes_supporting_lasy.rst]:
```suggestion
* `HiPACE++ <https://github.com/Hi-PACE/hipace>`_
```

File: docs/source/index.rst [hunk 8]
```diff
@@ -6,22 +6,37 @@
...
+The code is open-source and hosted on `github <https://github.com/LASY-org/lasy>`__. Contributions are welcome!
+
 .. panels::
     :card: + intro-card text-center
     :column: col-lg-6 col-md-6 col-sm-6 col-xs-12 p-2 d-flex
 
     ---
 
-    **User Guide**
+    **Getting Started**
```
Comment (MaxThevenet) [inline docs/source/index.rst:26]:
```suggestion
```

File: docs/source/code_overview/index.rst [hunk 9]
```diff
@@ -1,13 +1,55 @@
...
 
-What's this all about then?
+``lasy`` manipulates laser pulses, and operates on the laser envelope.
+It can be used to define the 3D profile of a laser pulse.
+The user can define seperately the transverse and longitudinal profile of the laser pulse either from a range common analytic profiles or using experimental measurements.
+Once defined the laser pulse may be propagated to a user defined location.
+Finally the laser profile may be outputted to file for use as an input to a variety of different simulation tools.
+
+Structure
+#########
```
Comment (MaxThevenet) [inline docs/source/code_overview/index.rst:11]:
```suggestion
#########

```

File: docs/source/index.rst [hunk 10]
```diff
@@ -6,22 +6,36 @@
 LASY |release| Documentation
 ============================
 
+.. warning::
+        Warning: This library is currently in development, and it is, at this stage, only meant to be used/tested by developers.
+        We plan on releasing the first version for general users (i.e. beta version) by summer 2023.
+
+
```
Comment (MaxThevenet) [inline docs/source/index.rst:13]:
```suggestion
        We plan on releasing the first version for general users (i.e. beta version) by summer 2023.
```

File: docs/source/index.rst (hunk 3 above)
Comment (MaxThevenet) [reply docs/source/index.rst:18]: I think it's fine now.

File: docs/source/index.rst [hunk 11]
```diff
@@ -57,7 +71,8 @@ LASY |release| Documentation
     **Tutorials**
     ^^^^^^^^^^^^^
 
-    Some step by step guides to using the code and some examples.
+    Some step-by-step guides to using the code and some common examples which you might find useful.
+    Some step-by-step guides to using the code and some common examples which you might find useful.
```
Comment (MaxThevenet) [inline docs/source/index.rst]:
```suggestion
```

File: docs/source/user_guide/index.rst [hunk 12]
```diff
@@ -1,10 +1,67 @@
...
+#############
```
Comment (MaxThevenet) [inline docs/source/user_guide/index.rst:21]:
```suggestion
#############

```

File: docs/source/index.rst (hunk 10 above)
Comment (MaxThevenet) [inline docs/source/index.rst]:
```suggestion
```

**Review by MaxThevenet — APPROVED** [review]

**Pull request merged**